src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,149 @@
 package core.basesyntax;
 
+import java.util.Map;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private int size;
+    private Node<K, V>[] table = new Node[DEFAULT_INITIAL_CAPACITY];
+
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int keyHashCode = key.hashCode();
+        int index = Math.abs(keyHashCode % table.length);
+
+        if (table[index] == null) {
+            table[index] = new Node<>(keyHashCode, key, value, null);
+            size++;
+        } else {
+            Node<K, V> currentNode = table[index];
+            while (true) {
+                if (currentNode.key != null && currentNode.key.equals(key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = new Node<>(keyHashCode, key, value, null);
+                    size++;
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+        }
 
+        if ((float) size / table.length > DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return getForNullKey();
+        }
+        int keyHashCode = key.hashCode();
+        int index = Math.abs(keyHashCode % table.length);
+        Node<K, V> currentNode = table[index];
+        if (currentNode == null) {
+            return null;
+        }
+        if (Objects.equals(currentNode.key,key)) {
+            return currentNode.value;
+        }
+        while (currentNode.next != null) {
+            currentNode = currentNode.next;
+            if (Objects.equals(currentNode.key,key)) {
+                return currentNode.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        table = new Node[oldTable.length * 2];
+        size = 0;
```

Comment:
Resetting the `size` to 0 during resizing is incorrect. The size should reflect the number of elements in the map, which does not change during resizing. Consider removing this line and adjusting the `put` method to not increment size during rehashing.